### ISSUE TITLE
Switching desktops no longer works on new Windows 11 update. Build 22631.3007

### ISSUE THREAD

**Issue by konekowo** [pmb6tz/windows-desktop-switcher#101]
Switching desktops doesn't work, but creating and deleting them works fine.
This is on the new Windows 11 update `2024-01 Cumulative Update Preview for Windows 11 Version 23H2 for x64-based Systems (KB5034204)` which was released yesterday on 1-23-2024

**Comment by cbs-cbs** [comment]
Ah! I've been attempting to troubleshoot this for like 45 minutes and was becoming exceedingly frustrated. Other scripts work fine. When something like this happens, is it likely someone figures it out and posts a fix here, relatively soon? [I'm rather new to AHK in general].

**Comment by konekowo** [comment]
Ok, looks like the people behind the VirtualDesktopAccessor.dll that this project uses had just fixed this in the latest release [https://github.com/Ciantic/VirtualDesktopAccessor/releases](https://github.com/Ciantic/VirtualDesktopAccessor/releases). So I guess until the owner of this repository updates the dll file on the repository, we're going to have to replace the dll file with the new one as a fix.

**Comment by cbs-cbs** [comment]
Can confirm - works again!

**Comment by ftanakasp** [comment]
Hi, sorry newbie here, can someone explain how can I replace the DLL? Thank you

**Comment by cbs-cbs** [comment]
Download it from the link provided above, figure out where you initially put the script - the original DLL should just be with it. Copy/paste the new one (and allow it to overwrite the old one), or simply delete the old one and move the new one to the folder.

**Comment by ftanakasp** [comment]
Thanks, I have done this (I check the DLL properties to make sure it was the new one, it is was the one created on the 29th), I tried even to fork a version of the repository and included the new dll download it and still not working, only able to delete and create desktops.

**Comment by cbs-cbs** [comment]
Ah weird. It was basically just plug-and-play for me. Did you do all the other obvious things (restart script, restart computer)? If that doesn't fix it, then it's already above my pay grade!

**Comment by ftanakasp** [comment]
yes restarted both, i tried even to restore the windows before the update. Thanks for the help. I will try one more time to restart.

**Comment by duluk** [comment]
Glad I checked here. Just updated Windows 11 and this script stopped working and I didn't realize how difficult I find using the OS without this script. (have it bound to a side button on my mouse for desktop profile; so easy to navigate desktops that I can't be bothered hitting a hotkey combination lol) Needless to say, I love this script!

**Comment by lukolszewski** [comment]
I can report that for me it broke following the below updates, but downloading the dll from the 3rd post fixed it again.
![image](https://github.com/pmb6tz/windows-desktop-switcher/assets/43611481/e8779a35-80ca-4183-9243-2a4f40a61870)

Windows is completely unusable without this functionality.

**Comment by ms4ndst** [comment]
Seems to be broken again in 23H2 22635.3350
Thing happens when running the script and trying to switch desktops.

**Comment by dillacorn** [comment]
Yep no longer working for me either...
I changed the "VirtaulDesktopAccessor.dll" to latest from that [repo](https://github.com/Ciantic/VirtualDesktopAccessor/releases) but no dice.

On version 23H2 ~ 22631.4037

Going to just rely on Win+Ctrl+Arrow keys for the time being......

I'd honestly have less of an issue with using taskview if it weren't so laggy/buggy occasionally.. also Windows 11s taskview IMO is a downgrade compared to 10s.. I can only see up to 6 workspaces and have to scroll to see the others... IT'S HONESTY TRASH..BUT SO WAS 10s so I guess I should stop complaining about that.

I wish someone could develop an actually good 3rd party taskview fork..

If it was actually good like GNOMEs taskview.. Hell even KDE Plasma 6 has good taskview adjacent functionality now I wouldn't have much of an issue using it.

**Comment by ms4ndst** [comment]
@dillacorn I agree. Tried Gwim, a great alternative with lots of tweak capabilities and super fast. If you wanna stick with windows taskviewer try turning off animations. It is really so much faster but not as pretty.

**Comment by dillacorn** [comment]
@ms4ndst are you talking about [glazewm](https://github.com/glzr-io/glazewm)?

I do like [i3](https://github.com/i3/i3) and [Sway](https://github.com/swaywm/sway) is my current preferred window manager.. I probably do need to give this a serious shot.

**Comment by ms4ndst** [comment]
No @dillacorn  I am more into Hyperland. "My other laptop" is running Arch with Hyperland using ml4w dotfiles. https://github.com/mylinuxforwork/dotfiles

I was talking about https://github.com/dalyIsaac/Whim

**Comment by SaminYaser-work** [comment]
Not working on 23H2 22631.3880. Fk microsoft

**Comment by skyfex** [comment]
For anyone needing a quick workaround, the VirtualDesktop command line tool here works for me: https://github.com/MScholtes/VirtualDesktop

Put the exe in the script folder and replace the content of desktop_switcher.ahk with
```
#Include %A_ScriptDir%\user_config.ahk
return

switchDesktopByNumber(targetDesktop)
{
	Run, "%A_ScriptDir%\VirtualDesktop11.exe" /Switch:%targetDesktop%,, Hide
}

MoveCurrentWindowToDesktop(desktopNumber) {
	Run, "%A_ScriptDir%\VirtualDesktop11.exe" /GetDesktop:%desktopNumber% /MoveActiveWindow /Switch,, Hide
}

```

That restores functionality for those two functions at least.. others can probably be implemented similarly
